Sources/Events/EventsManager.swift

```diff
@@ -158,6 +132,47 @@ actor EventsManager: EventsManagerType {
         return try await self.flushFeatureEventsInternal(batchSize: batchSize)
     }
 
+    func flushAdEvents(count: Int) async throws -> Int {
```

Comment:
Is there a specific reason the ad events are treated differently from any other event?

Reply:
Mainly they have their own queue and flush to a different endpoint. It's not clear if they should follow the same moments to trigger flushes compared to the rest of the events. We have had some back and forth with this in the past and ended up with this approach. Tagging @tonidero in case he has something more to add here.